Sources/ApodiniGRPC/GRPCInterfaceExporter.swift

```diff
@@ -1,231 +1,324 @@
+            ])
+        }
+    }
+    
+    
+    // MARK: Internal Stuff
+    
+    private func getMethodName<H>(for endpoint: Endpoint<H>) -> String {
+        if let methodName = endpoint[Context.self].get(valueFor: GRPCMethodNameContextKey.self) {
+            return methodName
+        } else {
+            // No explicit method name was specified, so we construct a default one based on the information we have about this handler.
+            // The problem is that we don't exactly have a lot of information about the handler.
+            // Essentially, there's only a handful of things we can use to construct a handler name:
+            // - path of the handler
+            // - handler type name (problematic w/ nested/generic handler types)
+            // - operation type (this would allow us to prepend verbs like "get" or "create"
+            // - communication pattern type (req-res, client-side stream, server-side stream, bidirectional stream). this is probaly the least useful one
+            let methodName = endpoint.absolutePath.reduce(into: "") { partialResult, pathComponent in
+                switch pathComponent {
+                case .root:
+                    break
+                case .string(let value):
+                    partialResult.append(value.capitalisingFirstCharacter)
+                case .parameter(let parameter):
+                    // TODO how should this be handled?
+                    // Maybe pull an objc and turn this into a "withX" schema?
+                    fatalError("TODO?")
+                }
             }
-        } catch GRPCServiceError.endpointAlreadyExists {
-            app.logger.error("Tried to overwrite endpoint \(methodName) for gRPC service \(serviceName)")
-        } catch {
-            app.logger.error("Error while exporting endpoint \(methodName) for gRPC service \(serviceName): \(error)")
+            // TODO if the path-derived method name is the "root" thing (eg: "V1"), somehow give it special handling!!!
+            return methodName
         }
     }
-
-    func retrieveParameter<Type: Decodable>(_ parameter: EndpointParameter<Type>, for request: GRPCMessage) throws -> Type?? {
-        guard let fieldTag = getFieldTag(for: parameter) else {
-            // If this occurs, something went fundamentally wrong in usage
-            // of the GRPC exporter.
-            // Each parameter should get a default field tag assigned
-            // above in the export() function.
-            fatalError("No default or explicit field tag available")
+    
+    
+    /// Registers some HTTP routes for accessing the proto reflection schema
+    private func setupReflectionHTTPRoutes() {
+        /// Make a JSON version of the whole gRPC schema available as a regular HTTP GET endpoint.
+        /// - NOTE this might not necessarily be the most dessirable thing, since it might expose internal data. But then again the OpenAPI interface exporter works the exact same way...
```

Comment:
```suggestion
        /// - NOTE this might not necessarily be the most desirable thing, since it might expose internal data. But then again the OpenAPI interface exporter works the exact same way...
```

Maybe it make sense to add a toggle for the reflection endpoints to the configuration 🤔

Reply:
Done. Added a config option to enable reflection (on by default)